lib/puppet/parser/functions/validate_nodejs_version.rb

```diff
@@ -0,0 +1,13 @@
+require 'semver'
+
+module Puppet::Parser::Functions
+  newfunction(:validate_nodejs_version) do |args|
+    version = SemVer.new(args[0])
+    if version < SemVer.new('v0.10.0')
+      raise Puppet::ParseError, ('All NodeJS versions below `v0.10.0` are not supported!')
+    end
+    if version < SemVer.new('v0.12.0')
```

Comment:
I think we should simply disallow all versions below 0.12 to avoid maintaining too much old stuff